Source/Libraries/openXDA.Model/Emails/EmailTypeBase.cs

```diff
@@ -29,13 +30,15 @@ public abstract class EmailTypeBase
     {
         [PrimaryKey(true)]
         public int ID { get; set; }
-
+        [ParentKey(typeof(EmailCategory))]
+        public int EmailCategoryID { get; set; }
+        [DefaultValue(false)]
+        public bool RequireApproval { get; set; }
+        public bool ShowSubscription { get; set; }
         public string Name { get; set; }
-
         public string Template { get; set; }
-
         public string FilePath { get; set; }
-
         public bool SMS { get; set; }
+        public string TriggerEmailSQL { get; set; }
```

Comment:
Perhaps the fact that this doesn't have a `DefaultValue` attribute despite the fact that the database does have a default value could be the primary reason why it was so easy to introduce broken email definitions with blank triggers. We have logic now to check this for null/empty, but perhaps this model should match the database definition anyway.
```suggestion
        [DefaultValue("SELECT 0")]
        public string TriggerEmailSQL { get; set; }
```